### PR TITLE
feat(Translational): add AccelerationSensor

### PR DESCRIPTION
**Pull request by sathvikbhagavan**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Adding AccelerationSensor which is particularly useful for measuring acceleration of masses without having to do `force/mass`.

### CODE REVIEW

**Review by baggepinnen — APPROVED** [review]

**Review by ven-k — APPROVED** [review]

**Comment by sathvikbhagavan** [comment]
@ChrisRackauckas, can this be merged?

File: test/Mechanical/translational.jl [hunk 1]
```diff
@@ -153,4 +153,24 @@ end
...
+        @named source = Sine(frequency = 2, amplitude = 1)
+        @named acc_output = RealOutput()
+        eqs = [
+            connect(force.f, source.output),
+            connect(force.flange, mass.flange),
+            connect(acc.flange, mass.flange),
+            connect(acc_output, acc.output)
+        ]
+        @named sys = ODESystem(eqs, t, [], []; systems = [force, source, mass, acc, acc_output])
+        s = complete(structural_simplify(sys))
```
Comment (ChrisRackauckas) [inline test/Mechanical/translational.jl:171]: Why not `@mtkbuild`?

Comment (sathvikbhagavan) [reply test/Mechanical/translational.jl:171]: I was just following the way other tests are written :sweat_smile:

**Comment by ChrisRackauckas** [comment]
we will need to mass update that

**Pull request merged**